.pre-commit-config.yaml

```diff
@@ -0,0 +1,33 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+  - repo: https://github.com/timothycrosley/isort
+    rev: 5.6.4
+    hooks:
+      - id: isort
+        files: .*.py
+  - repo: https://github.com/psf/black
+    rev: 20.8b1
+    hooks:
+      - id: black
+        args: ["--line-length", "79"]
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v3.2.0
+    hooks:
+      - id: check-ast
+      - id: check-case-conflict
+      - id: trailing-whitespace
```

Comment:
I'm not sure what filetypes this operates on, however, this will most likely corrupt tests that purposefully include trailing whitespace (because that's what they are testing for).

Perhaps we can add the trailing whitespace using `sed` for these tests to remove the problem.

Reply:
```
$ git grep '\\  *$' 
dockerfiles/cylc-remote/Dockerfile:    /bin/host-configure && \ 
tests/functional/validate/68-trailing_whitespace.t:            foo | bar \ 
```
The first one's an error!  Surprised that works :thinking:  I'll put up a fix, separately.

I've checked that the test file does get "corrected" by the hook.

Reply:
There's 1146 lines of inconsequential trailing whitespace.  `git grep '  *$'`

Reply:
> Surprised that works 🤔

Not too surprising, we added special logic to Cylc to protect against exactly the same issue!

Good only one test, should be able to convert that over to sed easy enough.

Reply:
I've pushed a sed-based fix for that test, to this branch.